fix: OPTIC-1982: Allow any errors to be displayed inline for external connection testing

bmartel opened the pull request:
This pull request modifies the error handling logic in the `ApiProvider` component to allow for more flexible inline error handling. The key change is the separation of the `setError` logic from the global error display logic.

### Error Handling Improvements:
* [`web/apps/labelstudio/src/providers/ApiProvider.tsx`](diffhunk://#diff-a19bbea04128de7fc2af68ba09aedf771c93e474e5b5323e3defa3c77ef9660eL229-R234): Introduced a conditional block to handle inline errors separately by calling `setError` before checking for global error display conditions. This ensures that inline error handling can occur independently of the global error logic.

bmartel committed:
fix: OPTIC-1982: Allow any errors to be displayed inline for external connection testing

AndrejOros commented:
/fm sync
> [Workflow run](https://github.com/HumanSignal/label-studio/actions/runs/14749144929)

AndrejOros committed:
Merge branch 'develop' into 'fb-optic-1982'

Workflow run: https://github.com/HumanSignal/label-studio/actions/runs/14749144929

AndrejOros committed:
Sync Follow Merge dependencies

Workflow run: https://github.com/HumanSignal/label-studio/actions/runs/14749144929

bmartel committed:
fixing the storage form handling of errors
